extension/encoding/awslogsencodingextension/unmarshaler/subscription-filter/subscription_filter.go

```diff
@@ -0,0 +1,140 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package subscriptionfilter // import "github.com/open-telemetry/opentelemetry-collector-contrib/extension/encoding/awslogsencodingextension/unmarshaler/subscription-filter"
```

Comment:
could you move this to an internal package?

Reply:
The reason this is public is so other components can use the `NewSubscriptionFilterUnmarshaler` function directly. For example, this would enable us to remove the [cwlog code in awsfirehosereceiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/awsfirehosereceiver/internal/unmarshaler/cwlog).

Rather than forcing users to configure an extension, there would be a special case where the receiver will instantiate the unmarshaler directly. We could also internally create an extension, but it seems a bit unnecessarily complicated.